src/Microsoft.Graph.Cli.Core/Authentication/EnvironmentCredential.cs

```diff
@@ -52,8 +52,8 @@ public EnvironmentCredential(string? tenantId, string? clientId, TokenCredential
         {
             _options = options ?? new TokenCredentialOptions();
 
-            tenantId ??= Environment.GetEnvironmentVariable(Utils.Constants.Environment.TenantId);
-            clientId ??= Environment.GetEnvironmentVariable(Utils.Constants.Environment.ClientId);
+            tenantId = Environment.GetEnvironmentVariable(Utils.Constants.Environment.TenantId) ?? tenantId;
```

Comment:
null coalesce operators with strings often lead to unexpected behavior, especially when NRT is turned on.
Have you considered ternaries with string.IsNullOrEmpty instead?